docs/mobile.md

```diff
@@ -0,0 +1,231 @@
+Accessibility id is recommended to use for locating element, as it rarely changed.
+
+* iOS uses [UIAccessibilityIdentification](https://developer.apple.com/documentation/uikit/uiaccessibilityidentification)
+* Android `accessibility id` matches the content-description
+
+Add `~` prefix to search for element by its accessibility id:
+
+```js
+I.seeElement('~startUserRegistrationButton');
+```
+
+Elements can also have ids, which can be located with `#` prefix.
+On Android it it is important to keep full package name in id locator:
+
+```js
+I.seeElement('#io.selendroid.testapp:id/inputUsername');
+```
+
+Elements can also be located by their class name. (For iOS they should start with `UIA`)
+
+```js
+I.fillField({ios: 'UIATextField', android: 'android.widget.EditText'}, '123456');
+```
+
+Buttons can be matched by their visible text:
+
+```js
+I.click('Click me!');
+```
+
+Native iOS/Android locators can be used with `android=` and `ios=` prefixes. [Learn more](http://webdriver.io/guide/usage/selectors.html#Mobile-Selectors).
+
+But how to get all those locators? We recommend to use [Appium Inspector](https://github.com/appium/appium-desktop).
+
+For Android you can use **UI Automator Viewer** bundled with Android SDK:
+
+![](https://user-images.githubusercontent.com/220264/27566310-1f631604-5aed-11e7-8b92-1ffe9e9f14f9.png)
+
+## Hybrid Apps and Contexts
+
+Mobile applications may have different contexts. For instance, there can be native view and web view with a browser instance in it.
+
+To execute commands in context of a webview use `within('webview')` function:
+
+```js
+I.click('~startWebView');
+within('webview', () => {
+  I.see('Preferred car');
+  I.click('Send me your name!');
+});
+```
+
+It will locate first available webview, switch to it, and switch back to native application after.
+Inside WebView all browser features are enabled: CSS locators, JavaScript, etc.
+
+To set a specific context use `{ web: 'webview.context' }` instead:
+
+```js
+within({webview: 'MyWEBVIEW_com.my.app'}, () => );
+```
+
+Alternatively use `switchToWeb` or `switchToNative` methods to switch between contexts.
+
+```js
+I.click('~startWebView');
+I.switchToWeb();
+I.see('Preferred car');
+I.click('Send me your name!');
+I.switchToNative();
+```
+
+To get a list of all contexts use `grabAllContexts` method:
+
+```js
+let contexts = yield I.grabAllContexts();
+```
+
+## Cross-Platform Testing
+
+It is often happen that mobile applications behave similarly on different platforms. Can we build on test for them? Yes!
+CodeceptJS provides a way to specify different locators for Android and iOS platforms:
+
+```js
+I.click({android: 'android.widget.Button', ios: '//UIAApplication[1]/UIAWindow[1]/UIAButton[1]'});
+```
+
+In case some code should be executed on one platform and ignored on others use `runOnAndroid` and `runOnIOS` methods:
+
+```js
+I.runOnAndroid(() => {
+  I.click('Hello Android');
+});
+I.runOnIOS(() => {
```

Comment:
We use the config informaition here, for sure. But steps from this function will be executed only if it is iOS.